src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -890,5 +891,655 @@ static void Main(string[] args)
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            A<Foo> aFooInstance = {|#4:new A<Foo>()|};
+        }
+    }
+
+class {|#1:A|}<T> where T : IFoo, new()
+{
+    public A()
+    {
+        IFoo foo = new T();
+        {|#0:foo.Bar()|};
+    }
+}
+
+class {|#2:Foo|} : IFoo
+{
+    public void {|#3:Bar|}() { }
+}
+
+[RequiresPreviewFeatures]
+interface IFoo
+{
+    void Bar();
+}
+}";
+
+            var test = TestCS(csInput);
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(0).WithArguments("Bar"));
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(1).WithArguments("A"));
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(2).WithArguments("Foo"));
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(3).WithArguments("Bar"));
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(4).WithArguments(".ctor"));
+            await test.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestCustomAttribute()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            A aObject = {|#0:new A()|};
+        }
+    }
+
+[My]
+class {|#1:A|}
+{
+}
+
+[RequiresPreviewFeatures]
+[AttributeUsage(AttributeTargets.All)]
+class MyAttribute : Attribute
+{
+}
+}";
+
+            var test = TestCS(csInput);
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(0).WithArguments(".ctor"));
+            await test.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestNestedInvocation()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine({|#0:A.B()|});
+        }
+    }
+
+class A
+{
+    [RequiresPreviewFeatures]
+    public static bool B() => true;
+}
+}";
+
+            var test = TestCS(csInput);
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(0).WithArguments("B"));
+            await test.RunAsync();
+
+        }
+
+        [Fact]
+        public async Task TestDebug()
```

Comment:
I assume we want to take this out once everything is working?

Reply:
Sorry, looks like the last commit (that deleted this) didn't get pushed. This is fixed now.